File: Leetcode/Javascript/easy/countMatches.js
var countMatches = function(items, ruleKey, ruleValue) {
    switch(ruleKey) {
      case "type":
      var ans=0  
      for(const i in items){
          if (items[i][0]==ruleValue){
            ans=ans+1
            }
        }
        break;
      case "color":
        var ans=0  
        for(const i in items){
            if (items[i][1]==ruleValue){
              ans=ans+1
              }
          }
  
        // code block
        break;
      case "name":
        var ans=0  
        for(const i in items){
            if (items[i][2]==ruleValue){
              ans=ans+1
              }
          }
  
      default:
        // code block
    }   
    return ans
  };
  
  console.log(countMatches([["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]],"color","silver"))

File: Leetcode/Python/easy/countMatches.py
class Solution:
    def countMatches(self, items, ruleKey, ruleValue):
        """
        items: List[List[str]], 
        ruleKey: str, 
        ruleValue: str 
        return: int
        """
        def first():
            cont=0
            for item in items:
                if item[0]==ruleValue:
                    cont+=1
            return cont
        def second():
            cont=0
            for item in items:
                if item[1]==ruleValue:
                    cont+=1
            return cont
        def third():
            cont=0
            for item in items:
                if item[2]==ruleValue:
                    cont+=1
            return cont
        
        options={"type":first,
        "color":second,
        "name":third,}
        return options[ruleKey]()

print(Solution().countMatches( [["phone","blue","pixel"],["computer","silver","lenovo"],["phone","gold","iphone"]],"color","silver"))